datadog_checks_dev/datadog_checks/dev/tooling/commands/release/testable.py

```diff
@@ -1,32 +1,51 @@
 # (C) Datadog, Inc. 2018-present
 # All rights reserved
 # Licensed under a 3-clause BSD style license (see LICENSE)
+import functools
 import time
+import typing
 
 import click
-from semver import parse_version_info
+import semver
 
 from ....subprocess import run_command
 from ....utils import basepath, chdir, get_next
 from ...constants import CHANGELOG_LABEL_PREFIX, CHANGELOG_TYPE_NONE, get_root
 from ...github import get_pr, get_pr_from_hash, get_pr_labels, get_pr_milestone, parse_pr_number
 from ...trello import TrelloClient
-from ...utils import format_commit_id, get_current_agent_version
+from ...utils import format_commit_id
 from ..console import CONTEXT_SETTINGS, abort, echo_failure, echo_info, echo_success, echo_waiting, echo_warning
 
 
-def validate_version(ctx, param, value):
-    if not value or ctx.resilient_parsing:
-        return
+def validate_version(ctx: click.Context, param: click.Parameter, value: str, ignore: typing.Sequence[str] = ()) -> str:
+    if value in ignore:
+        return value
 
-    try:
-        parts = value.split('.')
-        if len(parts) == 2:
-            parts.append('0')
-        version_info = parse_version_info('.'.join(parts))
-        return f'{version_info.major}.{version_info.minor}'
-    except ValueError:
-        raise click.BadParameter('needs to be in semver format x.y[.z]')
+    def parse_version(version: str) -> semver.VersionInfo:
+        try:
+            return semver.parse_version_info(version)
+        except ValueError:
+            raise click.BadParameter('needs to be in semver format M.m[.p[-r]]')
+
+    parts = value.split('.')
+
+    if len(parts) == 2:
+        example_release_tag = f'{value}.1'
+        example_release_branch = f'{value}.x'
+        raise click.BadParameter(
+            f"Using a minor version ({value!r}) is ambiguous. Use a release tag (e.g. {example_release_tag!r}) "
+            f"or a release branch (e.g. {example_release_branch!r}) instead."
+        )
+
+    if len(parts) == 3 and parts[-1] == 'x':
+        # Treat as a release branch, e.g. '7.17.x'.
+        parts[2] = '0'
+        version_info = parse_version('.'.join(parts))
+        return f'{version_info.major}.{version_info.minor}.x'
+
+    # Treat as a fully-resolved, semver-compliant version string, e.g. '7.17.0', or '7.17.0-rc.2'.
+    version_info = parse_version('.'.join(parts))
+    return str(version_info)
```

Comment:
I think we can get rid of the `validate_version()`. The diff command will fail later naturally if the references are wrong.

Reply:
Yup, not necessary anymore!